routing/additionaledge.go

```diff
@@ -0,0 +1,113 @@
+package routing
+
+import (
+	"errors"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightningnetwork/lnd/channeldb/models"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing/route"
+)
+
+var (
+	// ErrNoPayLoadSizeFunc is returned when no payload size function is
+	// definied.
+	ErrNoPayLoadSizeFunc = errors.New("no payloadSizeFunc definied for " +
+		"additional edge")
+)
+
+// AdditionalEdge is an interface which specifies additional edges which can
+// be appended to an existing route. Compared to normal edges of a route they
+// provide an explicit payload size function and are introduced because blinded
+// paths differ in their payload structure.
+type AdditionalEdge interface {
+	// PayloadSize returns the size of the payload for the additional edge.
+	PayloadSize(amount lnwire.MilliSatoshi, expiry uint32, legacy bool,
+		channelID uint64) uint64
+
+	// EdgePolicy returns the policy of the additional edge.
+	EdgePolicy() *models.CachedEdgePolicy
+}
+
+// PayloadSizeFunc defines the interface for the payload size function.
+type PayloadSizeFunc func(amount lnwire.MilliSatoshi, expiry uint32,
```

Comment:
Unfortunately yes because the `unifiedEdge`  needs the exact amount and htlc_timeout value to calculate the exact size of the payload which we only have available during building up the route. We cannot just use a default size because the values are truncated integer types:

```
type: 2 (amt_to_forward)
data:
[tu64:amt_to_forward]
type: 4 (outgoing_cltv_value)
data:
[tu32:outgoing_cltv_value]
```